dataset_loaders/images/polyps912.py

```diff
@@ -87,12 +90,46 @@ def __init__(self, which_set='train', *args, **kwargs):
         else:
             raise ValueError("Unknown set requested: %s" % which_set)
 
+        self.preload = preload
+
         # Define the images and mask paths
         self.image_path = os.path.join(self.path, self.which_set, 'images')
         self.mask_path = os.path.join(self.path, self.which_set, 'masks2')
 
+        if self.preload:
+            self._preload_data()
+
         super(Polyps912Dataset, self).__init__(*args, **kwargs)
 
+    def _preload_data(self):
+        """Preload all data in memory.
+
+        The images will be stored as a list of ndarrays in self.image,
+        the masks will be in self.mask, in the same order as
+        self.filename.
+        In addition, self.image_name_to_idx will contain a dictionary
+        mapping the root of the image name to its index.
+        """
+        from skimage import io
+        image_all = []
+        mask_all = []
+        image_name_to_idx = {}
+        for idx, img_name in enumerate(self.filenames):
+            image_name_to_idx[img_name] = idx
+            img = io.imread(os.path.join(self.image_path, img_name + ".bmp"))
```

Comment:
Sure, I'll refactor the loading code for that dataset.
I'm not familiar enough with the rest of the code to have a good sense of what can be refactored in parallel_loader.py, and what needs to be done in each dataset, and unfortunately I will not have time to dive into that soon.